### ISSUE TITLE
Error in IDE "Failed to check if current file is included"

### ISSUE THREAD

**Issue by Fischenschaftler** [SchwarzIT/spectral-intellij-plugin#89]
When starting the Plugin in IntelliJ IDEA 2023.2 (Community Edition), this error is displayed:

Stacktraces:

```
java.lang.Throwable: Failed to check if current file is included
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:370)
	at com.schwarzit.spectralIntellijPlugin.SpectralExternalAnnotator.collectInformation(SpectralExternalAnnotator.kt:34)
	at com.schwarzit.spectralIntellijPlugin.SpectralExternalAnnotator.collectInformation(SpectralExternalAnnotator.kt:21)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.collectInformationWithProgress(ExternalToolPass.java:131)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:95)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:57)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:390)
	at com.intellij.platform.diagnostic.telemetry.helpers.TraceKt.runWithSpanIgnoreThrows(trace.kt:77)
	at com.intellij.platform.diagnostic.telemetry.helpers.TraceUtil.runWithSpanThrows(TraceUtil.java:24)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:386)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1133)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$3(PassExecutorService.java:377)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:376)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:352)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:201)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:207)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:350)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:185)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)

```

**Comment by markbrockhoff** [comment]
Hi @Fischenschaftler & @christian-esken, thanks for the report.
Could you provide more information about your environment the issue occured in? Maybe even a way to reproduce it?
Especially your settings for the Spectral plugin would be interresting.

Version 2.1.2 of the plugin also features an enhanced error handling in this area, so if the issue still occurs with that version it might give some more information about what's going on.

**Comment by Fischenschaftler** [comment]
I'm using IntelliJ IntelliJ IDEA 2023.2 (Community Edition):
Build #IC-232.8660.185
Runtime version: 17.0.7+7-b1000.6 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0
Kotlin: 232-1.9.0-IJ8660.185 

In the plugin settings, the path to the ruleset and the file extension of the files to be checked are specified. 
Different rulesets have been tried out. Spectral CLI is installed and can be called from the command line.

**Comment by markbrockhoff** [comment]
Ok, from the stacktrace I suspect the issue to be caused by our path matching algorithm.
Could you possibly give concrete values for the path of the file you lint, the base path of your project and the value set for the paths to include inside the settings?
Then I can try to reproduce the issue and possibly even add a test case for that.

**Comment by Fischenschaftler** [comment]
The path to the Ruleset is C:\Users\Username\Projekt\src\main\ruleset\ruleset.yaml
The included files to lint are: **.yaml
For the CLI, linting works using:
spectral lint < C:\Users\Username\Projekt\API_Name.yaml> --ruleset <C:\Users\Username\Projekt\src\main\ruleset\ruleset.yaml>

**Comment by longbowww** [comment]
This error only presents on windows for me, on macOS it works fine.

At the moment is unclear to me, if the error is just masking an underlying problem, as calling the spectral linting command manually will yield a policy block error (company device).

**Comment by markbrockhoff** [comment]
Hi,

we just released version 3 of the plugin. The new version includes a complete rewrite of the path matching logic, now using ant pattern matching. This should enable support for windows. Please feel free to give it a try.

I'd close this issue meanwhile but if the problem still persists, feel free to let us know.

**Comment by tintin92350** [comment]
I have the same issue on version 3.0.0:

![image](https://github.com/SchwarzIT/spectral-intellij-plugin/assets/25781206/0cfcc7c5-7543-47b7-9df2-6f16ec377ee1)

**Comment by tintin92350** [comment]
The error come from this code:
```kotlin
if (!Paths.get(s).isAbsolute) {
                var base = basePath.toString()
                if (!base.endsWith(separator)) base += separator
                globPattern = base + s
            }
```

You are checking if `s` is absolute but if it contains `*` at the beginning it will fail by the expected exception:
```
Illegal char <*> at index 0
```

So we need to find a way to check absolute path for only path not beginning by a pattern

**Comment by markbrockhoff** [comment]
Closing this issue as it should be fixed by #106 
If the issue is still occuring, feel free to re-open it.
